x-pack/plugins/data_enhanced/server/search/session/session_service.ts

```diff
@@ -217,6 +219,7 @@ export class SearchSessionService
       restoreState = {},
     }: Partial<SearchSessionSavedObjectAttributes>
   ) => {
+    if (!this.sessionConfig.enabled) throw new Error('Feature is disabled');
```

Comment:
Nit: May want to add the name of the feature to avoid ambiguous error logs:

```suggestion
    if (!this.sessionConfig.enabled) throw new Error('Search sessions are disabled');
```